mcs/tools/corcompare/mono-api-info.cs

```diff
@@ -196,25 +317,27 @@ public void DoOutput ()
 
 		AssemblyDefinition LoadAssembly (string assembly)
 		{
-			try {
-				if (File.Exists (assembly))
-					return TypeHelper.Resolver.ResolveFile (assembly);
+			if (File.Exists (assembly))
+				return state.TypeHelper.Resolver.ResolveFile (assembly);
 
-				return TypeHelper.Resolver.Resolve (AssemblyNameReference.Parse (assembly), new ReaderParameters ());
-			} catch (Exception e) {
-				Console.WriteLine (e);
-				return null;
-			}
+			return state.TypeHelper.Resolver.Resolve (AssemblyNameReference.Parse (assembly), new ReaderParameters ());
```

Comment:
No use catching the exception here. If this assembly cannot be loaded, then the very assembly we are trying to generate info for is not loaded. If the assembly is not loaded, nothing will be generated. If nothing is generated, then why do we even need this tool? If we wanted something to be generated, then throw an error so we know it failed. #Logic